block managed_namespace_metadata panic

```
    sync_policy {

      managed_namespace_metadata {
        labels      = each.value.managed_namespace_metadata_labels
        annotations = each.value.managed_namespace_metadata_annotations
      }
    }
```

```
variable "multiple_sources_app" {
  type = map(object({
    managed_namespace_metadata_labels      = optional(map(string))
    managed_namespace_metadata_annotations = optional(map(string))
  }))
  default = {}
}
```

```
│ Error: Plugin did not respond
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin6.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵
╷
│ Error: Request cancelled
│ 
│ The plugin6.(*GRPCProvider).UpgradeResourceState request was cancelled.
╵
╷
│ Error: Plugin did not respond
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin6.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵
╷
│ Error: Request cancelled
│ 
│ The plugin6.(*GRPCProvider).UpgradeResourceState request was cancelled.
╵
╷
│ Error: Request cancelled
│ 
│ The plugin6.(*GRPCProvider).UpgradeResourceState request was cancelled.
╵
╷
│ Error: Plugin did not respond
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin6.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵
╷
│ Error: Plugin did not respond
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin6.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵
╷
│ Error: Plugin did not respond
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin6.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵
╷
│ Error: Plugin did not respond
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin6.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵
╷
│ Error: Request cancelled
│ 
│ The plugin6.(*GRPCProvider).UpgradeResourceState request was cancelled.
╵
╷
│ Error: Plugin did not respond
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin6.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵
╷
│ Error: Plugin did not respond
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin6.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵
╷
│ Error: Plugin did not respond
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin6.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵
╷
│ Error: Plugin did not respond
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin6.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵
Stack trace from the terraform-provider-argocd_v7.0.3 plugin:
panic: interface conversion: interface {} is nil, not map[string]interface {}
goroutine 122 [running]:
github.com/oboukili/terraform-provider-argocd/argocd.expandApplicationSyncPolicy({0x2f7e060, 0xc0023a0bd0})
	github.com/oboukili/terraform-provider-argocd/argocd/structure_application.go:407 +0xa65
github.com/oboukili/terraform-provider-argocd/argocd.expandApplicationSpec(0x0?)
	github.com/oboukili/terraform-provider-argocd/argocd/structure_application.go:43 +0x213
github.com/oboukili/terraform-provider-argocd/argocd.expandApplication(_)
	github.com/oboukili/terraform-provider-argocd/argocd/structure_application.go:16 +0x10d
github.com/oboukili/terraform-provider-argocd/argocd.resourceArgoCDApplicationUpdate({0x3c00[320](https://git.sbercloud.tech/dsworks/infrastructure/-/jobs/7370002#L320)?, 0xc000038620}, 0xc002377e00, {0x32d5ec0?, 0xc000828a00?})
	github.com/oboukili/terraform-provider-argocd/argocd/resource_argocd_application.go:262 +0x325
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0xc0007b3b20, {0x3c00278, 0xc00159f290}, 0xd?, {0x32d5ec0, 0xc000828a00})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/resource.go:840 +0x11b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0007b3b20, {0x3c00278, 0xc00159f290}, 0xc000faa9c0, 0xc000ef8980, {0x32d5ec0, 0xc000828a00})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/resource.go:947 +0x83a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc000981e60, {0x3c00278?, 0xc00159f0e0?}, 0xc000f44000)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/grpc_provider.go:1153 +0xdbc
github.com/hashicorp/terraform-plugin-mux/tf5to6server.v5tov6Server.ApplyResourceChange({{0x3c1b840?, 0xc000981e60?}}, {0x3c00278, 0xc00159f0e0}, 0x0?)
	github.com/hashicorp/terraform-plugin-mux@v0.16.0/tf5to6server/tf5to6server.go:38 +0x54
github.com/hashicorp/terraform-plugin-mux/tf6muxserver.(*muxServer).ApplyResourceChange(0x3c002b0?, {0x3c00278?, 0xc00159ede0?}, 0xc000c01f90)
	github.com/hashicorp/terraform-plugin-mux@v0.16.0/tf6muxserver/mux_server_ApplyResourceChange.go:36 +0x193
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ApplyResourceChange(0xc0009261e0, {0x3c00278?, 0xc00159e180?}, 0xc00076fdc0)
	github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov6/tf6server/server.go:865 +0x3d0
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ApplyResourceChange_Handler({0x[340](https://git.sbercloud.tech/dsworks/infrastructure/-/jobs/7370002#L340)da20?, 0xc0009261e0}, {0x3c00278, 0xc00159e180}, 0xc0014dc480, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:518 +0x169
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000042600, {0x3c00278, 0xc00159e0c0}, {0x3c0f940, 0xc0002ba180}, 0xc001c12ea0, 0xc000da2900, 0x5904858, 0x0)
	google.golang.org/grpc@v1.63.2/server.go:1369 +0xe23
google.golang.org/grpc.(*Server).handleStream(0xc000042600, {0x3c0f940, 0xc0002ba180}, 0xc001c12ea0)
	google.golang.org/grpc@v1.63.2/server.go:1780 +0x1016
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	google.golang.org/grpc@v1.63.2/server.go:1019 +0x8b
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 78
	google.golang.org/grpc@v1.63.2/server.go:1030 +0x135
Error: The terraform-provider-argocd_v7.0.3 plugin crashed!
This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
Cleaning up project directory and file based variables
00:00
ERROR: Job failed: command terminated with exit code 1
```

```
+ create
Terraform will perform the following actions:
  # module.argocd-resources.argocd_application.multiple_sources["gitlab-runner-infra"] will be created
  + resource "argocd_application" "multiple_sources" {
      + cascade  = true
      + id       = (known after apply)
      + status   = (known after apply)
      + validate = true
      + wait     = false
      + metadata {
          + generation       = (known after apply)
          + name             = "gitlab-runner-infra"
          + namespace        = "argocd"
          + resource_version = (known after apply)
          + uid              = (known after apply)
        }
      + spec {
          + project                = "default"
          + revision_history_limit = 10
          + destination {
                name      = null
              + namespace = "gitlab-runner"
              + server    = "https://kubernetes.default.svc"
            }
          + source {
              + chart           = "gitlab-runner"
              + path            = "."
              + repo_url        = "http://charts.gitlab.io/"
              + target_revision = "0.70.1"
              + helm {
                  + value_files = [
                      + null,
                    ]
                }
            }
          + sync_policy {
              + sync_options = [
                  + "CreateNamespace=true",
                ]
              + automated {
                  + prune     = true
                  + self_heal = true
                }
              + managed_namespace_metadata {
                  + annotations = {
                      + "argocd.argoproj.io/managed-by" = "argocd"
                    }
                  + labels      = {
                      + "argocd.argoproj.io/managed-by" = "argocd"
                    }
                }
              + retry {
                  + limit = "5"
                  + backoff {
                      + duration     = "5s"
                      + factor       = "2"
                      + max_duration = "3m"
                    }
                }
            }
        }
      + timeouts {}
    }
Plan: 1 to add, 0 to change, 0 to destroy.
module.argocd-resources.argocd_application.multiple_sources["gitlab-runner-infra"]: Creating...
╷
│ Warning: Value for undeclared variable
│ 
│ The root module does not declare a variable named "cloudflare_token" but a
│ value was found in file ".secrets.auto.tfvars". If you meant to use this
│ value, add a "variable" block to the configuration.
│ 
│ To silence these warnings, use TF_VAR_... environment variables to provide
│ certain "global" settings to all configurations in your organization. To
│ reduce the verbosity of these warnings, use the -compact-warnings option.
╵
╷
│ Error: Plugin did not respond
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin6.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵
Stack trace from the terraform-provider-argocd_v7.0.3 plugin:
panic: interface conversion: interface {} is nil, not string
goroutine 95 [running]:
github.com/oboukili/terraform-provider-argocd/argocd.expandApplicationSourceHelm({0xc000fe4130, 0xc000fd3380?, 0x3534eed?})
	github.com/oboukili/terraform-provider-argocd/argocd/structure_application.go:250 +0x994
github.com/oboukili/terraform-provider-argocd/argocd.expandApplicationSource({0xc000fe40d0, 0x1, 0x3537ddd?})
	github.com/oboukili/terraform-provider-argocd/argocd/structure_application.go:88 +0x29d
github.com/oboukili/terraform-provider-argocd/argocd.expandApplicationSpec(0x0?)
	github.com/oboukili/terraform-provider-argocd/argocd/structure_application.go:54 +0x33f
github.com/oboukili/terraform-provider-argocd/argocd.expandApplication(_)
	github.com/oboukili/terraform-provider-argocd/argocd/structure_application.go:16 +0x10d
github.com/oboukili/terraform-provider-argocd/argocd.resourceArgoCDApplicationCreate({0x3c00320?, 0xc0003bb960}, 0x0?, {0x32d5ec0?, 0xc0001b8500})
	github.com/oboukili/terraform-provider-argocd/argocd/resource_argocd_application.go:86 +0xe8
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc000641b20, {0x3c00278, 0xc000ecccf0}, 0xd?, {0x32d5ec0, 0xc0001b8500})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/resource.go:806 +0x11b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000641b20, {0x3c00278, 0xc000ecccf0}, 0xc000d01520, 0xc0002a1e80, {0x32d5ec0, 0xc0001b8500})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/resource.go:937 +0xa89
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc000a01890, {0x3c00278?, 0xc000eccb40?}, 0xc0003340f0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.34.0/helper/schema/grpc_provider.go:1153 +0xdbc
github.com/hashicorp/terraform-plugin-mux/tf5to6server.v5tov6Server.ApplyResourceChange({{0x3c1b[84](https://git.sbercloud.tech/dsworks/infrastructure/-/jobs/7376317#L84)0?, 0xc000a01890?}}, {0x3c00278, 0xc000eccb40}, 0x0?)
	github.com/hashicorp/terraform-plugin-mux@v0.16.0/tf5to6server/tf5to6server.go:38 +0x54
github.com/hashicorp/terraform-plugin-mux/tf6muxserver.(*muxServer).ApplyResourceChange(0x3c002b0?, {0x3c00278?, 0xc000ecc840?}, 0xc0003340a0)
	github.com/hashicorp/terraform-plugin-mux@v0.16.0/tf6muxserver/mux_server_ApplyResourceChange.go:36 +0x193
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ApplyResourceChange(0xc00099fb80, {0x3c00278?, 0xc000e97e30?}, 0xc0000f27e0)
	github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov6/tf6server/server.go:865 +0x3d0
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ApplyResourceChange_Handler({0x340da20?, 0xc00099fb80}, {0x3c00278, 0xc000e97e30}, 0xc0002a1200, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:518 +0x169
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000184600, {0x3c00278, 0xc000e97da0}, {0x3c0f940, 0xc000982000}, 0xc000ec4a20, 0xc000d9c930, 0x5904[85](https://git.sbercloud.tech/dsworks/infrastructure/-/jobs/7376317#L85)8, 0x0)
	google.golang.org/grpc@v1.63.2/server.go:1369 +0xe23
google.golang.org/grpc.(*Server).handleStream(0xc000184600, {0x3c0f940, 0xc000982000}, 0xc000ec4a20)
	google.golang.org/grpc@v1.63.2/server.go:1780 +0x1016
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	google.golang.org/grpc@v1.63.2/server.go:1019 +0x8b
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 40
	google.golang.org/grpc@v1.63.2/server.go:1030 +0x135
Error: The terraform-provider-argocd_v7.0.3 plugin crashed!
This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
Cleaning up project directory and file based variables
00:01
ERROR: Job failed: command terminated with exit code 1
```

block retry {}

```
Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # module.argocd-resources.argocd_application.application["grafana-test"] will be created
  + resource "argocd_application" "application" {
      + cascade  = true
      + id       = (known after apply)
      + status   = (known after apply)
      + validate = true
      + wait     = false

      + metadata {
          + generation       = (known after apply)
          + name             = "grafana-test"
          + namespace        = "argocd"
          + resource_version = (known after apply)
          + uid              = (known after apply)
        }

      + spec {
          + project                = "default"
          + revision_history_limit = 10

          + destination {
              + namespace = "test-deploy"
              + server    = "https://kubernetes.default.svc"
            }

          + source {
              + chart           = "grafana"
              + path            = "."
              + repo_url        = "https://grafana.github.io/helm-charts"
              + target_revision = "8.5.2"

              + helm {
                  + value_files = [
                      + "$values/values/grafana.yaml",
                    ]
                }
            }
          + source {
              + path            = "."
              + ref             = "values"
              + repo_url        = "https://git.sbercloud.tech/dsworks/infrastructure.git"
              + target_revision = "HEAD"
            }

          + sync_policy {
              + sync_options = [
                  + "CreateNamespace=true",
                ]

              + automated {
                  + prune     = true
                  + self_heal = true
                }

              + retry {}
            }
        }

      + timeouts {}
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

module.argocd-resources.argocd_application.application["grafana-test"]: Creating...
╷
│ Error: Plugin did not respond
│ 
│ The plugin encountered an error, and failed to respond to the plugin6.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain
│ more details.
╵

Stack trace from the terraform-provider-argocd_v7.2.0 plugin:

panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 145 [running]:
github.com/argoproj-labs/terraform-provider-argocd/argocd.expandApplicationSyncPolicy({0x10552d240, 0x140009cad80})
        github.com/argoproj-labs/terraform-provider-argocd/argocd/structure_application.go:366 +0x9bc
github.com/argoproj-labs/terraform-provider-argocd/argocd.expandApplicationSpec(0x140009ca0f0)
        github.com/argoproj-labs/terraform-provider-argocd/argocd/structure_application.go:43 +0x1f0
github.com/argoproj-labs/terraform-provider-argocd/argocd.expandApplication(_)
        github.com/argoproj-labs/terraform-provider-argocd/argocd/structure_application.go:16 +0xcc
github.com/argoproj-labs/terraform-provider-argocd/argocd.resourceArgoCDApplicationCreate({0x105b52180, 0x140009005b0}, 0x14000fa7f00, {0x1058a0680, 0x140000f8508})
        github.com/argoproj-labs/terraform-provider-argocd/argocd/resource_argocd_application.go:86 +0x94
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x1400099d960, {0x105b520d8, 0x140008f53b0}, 0x14000fa7f00, {0x1058a0680, 0x140000f8508})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.35.0/helper/schema/resource.go:806 +0xe4
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0x1400099d960, {0x105b520d8, 0x140008f53b0}, 0x14000313e10, 0x14001245080, {0x1058a0680, 0x140000f8508})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.35.0/helper/schema/resource.go:937 +0x884
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0x14000624b88, {0x105b520d8?, 0x140008f4ed0?}, 0x140000d5180)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.35.0/helper/schema/grpc_provider.go:1155 +0xaa4
github.com/hashicorp/terraform-plugin-mux/tf5to6server.v5tov6Server.ApplyResourceChange({{0x105b70bd8?, 0x14000624b88?}}, {0x105b520d8, 0x140008f4ed0}, 0x0?)
        github.com/hashicorp/terraform-plugin-mux@v0.17.0/tf5to6server/tf5to6server.go:38 +0x58
github.com/hashicorp/terraform-plugin-mux/tf6muxserver.(*muxServer).ApplyResourceChange(0x140005fe500, {0x105b520d8?, 0x140008f4c00?}, 0x140000d5130)
        github.com/hashicorp/terraform-plugin-mux@v0.17.0/tf6muxserver/mux_server_ApplyResourceChange.go:36 +0x184
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ApplyResourceChange(0x140004dd400, {0x105b520d8?, 0x140008f4210?}, 0x14000490000)
        github.com/hashicorp/terraform-plugin-go@v0.25.0/tfprotov6/tf6server/server.go:865 +0x2b4
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ApplyResourceChange_Handler({0x105a10be0, 0x140004dd400}, {0x105b520d8, 0x140008f4210}, 0x14001244380, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.25.0/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:611 +0x1c0
google.golang.org/grpc.(*Server).processUnaryRPC(0x140001b6a00, {0x105b520d8, 0x140008f4180}, {0x105b63b80, 0x14000dc9520}, 0x14001384000, 0x14000efd200, 0x1078c3b18, 0x0)
        google.golang.org/grpc@v1.67.1/server.go:1394 +0xb78
google.golang.org/grpc.(*Server).handleStream(0x140001b6a00, {0x105b63b80, 0x14000dc9520}, 0x14001384000)
        google.golang.org/grpc@v1.67.1/server.go:1805 +0xb20
google.golang.org/grpc.(*Server).serveStreams.func2.1()
        google.golang.org/grpc@v1.67.1/server.go:1029 +0x8c
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 102
        google.golang.org/grpc@v1.67.1/server.go:1040 +0x13c

Error: The terraform-provider-argocd_v7.2.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

--- a/argocd/structure_application.go
+++ b/argocd/structure_application.go
@@ -469,15 +469,16 @@ func expandApplicationSyncPolicy(sp interface{}) (*application.SyncPolicy, error
 	}
 
 	if _mnm, ok := p["managed_namespace_metadata"].([]interface{}); ok && len(_mnm) > 0 {
-		mnm := _mnm[0].(map[string]interface{})
-		syncPolicy.ManagedNamespaceMetadata = &application.ManagedNamespaceMetadata{}
+		if mnm, ok := _mnm[0].(map[string]interface{}); ok {
+			syncPolicy.ManagedNamespaceMetadata = &application.ManagedNamespaceMetadata{}
 
-		if a, ok := mnm["annotations"]; ok {
-			syncPolicy.ManagedNamespaceMetadata.Annotations = expandStringMap(a.(map[string]interface{}))
-		}
+			if a, ok := mnm["annotations"]; ok {
+				syncPolicy.ManagedNamespaceMetadata.Annotations = expandStringMap(a.(map[string]interface{}))
+			}
 
-		if l, ok := mnm["labels"]; ok {
-			syncPolicy.ManagedNamespaceMetadata.Labels = expandStringMap(l.(map[string]interface{}))
+			if l, ok := mnm["labels"]; ok {
+				syncPolicy.ManagedNamespaceMetadata.Labels = expandStringMap(l.(map[string]interface{}))
+			}
 		}
 	}
 

